src/psyclone/tests/psyir/transformations/matmul2code_trans_test.py

```diff
@@ -0,0 +1,389 @@
+    assert ("The supplied node should be a BinaryOperation but found "
+            "'NoneType'." in str(excinfo.value))
+
+
+def test_validate2():
+    '''Check that the Matmul2Code validate method raises the expected
+    exception when the supplied node is a binary operation but not a
+    MATMUL.
+
+    '''
+    trans = Matmul2CodeTrans()
+    with pytest.raises(GenerationError) as excinfo:
+        trans.validate(BinaryOperation.create(
+            BinaryOperation.Operator.ADD, Literal("1.0", DataType.REAL),
+            Literal("1.0", DataType.REAL)))
+    assert ("Generation Error: The supplied node should be a MATMUL "
+            "BinaryOperation but found 'Operator.ADD'." in str(excinfo.value))
+
+
+def test_validate3():
+    '''Check that the Matmul2Code validate method raises the expected
+    exception when the supplied node is a MATMUL binary operation but
+    it is not the only operation on the RHS of an assignment.
+
+    '''
+    trans = Matmul2CodeTrans()
+    array = Array.create(DataSymbol("x", DataType.REAL, shape=[10]),
+                         [Literal("10", DataType.INTEGER)])
+    matmul = BinaryOperation.create(
+        BinaryOperation.Operator.MATMUL, array, array)
+    _ = BinaryOperation.create(BinaryOperation.Operator.MUL, matmul,
+                               Literal("1.0", DataType.REAL))
+    with pytest.raises(GenerationError) as excinfo:
+        trans.validate(matmul)
+    assert ("Generation Error: Matmul2CodeTrans only supports the "
+            "transformation of a MATMUL operation when it is the sole "
+            "operation on the rhs of an assignment." in str(excinfo.value))
+
+
+def test_validate4():
+    '''Check that the Matmul2Code validate method raises the expected
+    exception when the supplied node is a MATMUL binary operation but
+    either or both arguments are not references.
+
+    '''
+    trans = Matmul2CodeTrans()
+    array = Array.create(DataSymbol("x", DataType.REAL, shape=[10]),
+                         [Literal("10", DataType.INTEGER)])
+    mult = BinaryOperation.create(
+        BinaryOperation.Operator.MUL, array, array)
+    matmul = BinaryOperation.create(
+        BinaryOperation.Operator.MATMUL, mult, mult)
+    _ = Assignment.create(array, matmul)
+    with pytest.raises(GenerationError) as excinfo:
+        trans.validate(matmul)
+    assert ("Expected children of a MATMUL BinaryOperation to be references, "
+            "but found 'BinaryOperation', 'BinaryOperation'."
+            in str(excinfo.value))
+
+
+def test_validate5():
+    '''Check that the Matmul2Code validate method raises the expected
+    exception when the supplied node is a MATMUL binary operation but
+    either or both of its arguments are references to datasymbols that
+    are not arrays.
+
+    '''
+    trans = Matmul2CodeTrans()
+    scalar = Reference(DataSymbol("x", DataType.REAL))
+    matmul = BinaryOperation.create(
+        BinaryOperation.Operator.MATMUL, scalar, scalar)
```

Comment:
Again, this should really be picked-up by the create() operation.

Reply:
As above. Should I create an issue and reference it?

Reply:
Yes please.